src/tools/compiletest/src/common.rs

```diff
@@ -456,11 +457,17 @@ pub enum Endian {
 }
 
 impl TargetCfg {
-    fn new(rustc_path: &Path, target: &str) -> TargetCfg {
+    fn new(rustc_path: &Path, target: &str, target_rustcflags: &Option<Vec<String>>) -> TargetCfg {
+        let mut target_rustcflags_strs = Vec::<&str>::new();
+        if let Some(i) = target_rustcflags {
+            target_rustcflags_strs = i.iter().map(String::as_str).collect();
```

Comment:
Are you sure this is needed? Command::args takes an iterator of `Item = impl AsRef<OsStr>` and `String: AsRef<OsStr>`. Your iterator will be of `&String`, but there's a blanket `impl<T, U> AsRef<U> for &T where T: AsRef<U>` so that should be ok?

Reply:
I'll revisit this implementation -- a lot of this complication came from the `Option<Vec<String>>` type. With the switch to `Vec<String>`, this may become much cleaner than we are seeing here